hub/codec/codecs.py

```diff
@@ -0,0 +1,380 @@
+
+        Args:
+            bytes_ (bytes): Encoded images and metadata.
+
+        Returns:
+            Decoded image or array with multiple images.
+        """
+        data = self._msgpack.decode(buf)[0]
+        if "items_shape" not in data:
+            images = self.decode_single_image(data["items"])
+        else:
+            items = data["items"]
+            images = np.zeros(
+                data["items_shape"] + data["image_shape"], dtype=data["dtype"]
+            )
+
+            for i, index in enumerate(np.ndindex(tuple(data["items_shape"]))):
+                images[index] = self.decode_single_image(items[i])
+
+        if data.get("append_one"):
+            images = np.reshape(images, images.shape + (1,))
+        return images
+
+    def get_config(self):
+        """Get compressor configuration dict"""
+        return {"id": self.codec_id, "single_channel": self.single_channel}
+
+
+class JpegCodec(BaseImgCodec, Codec):
+    """Jpeg compressor for image data"""
+
+    def __init__(self, single_channel: bool = True):
+        """
+        Initialize Jpeg compressor
+
+        Args:
+            single_channel (bool): if True, encoder will remove the last dimension of input if it is 1.
+        """
+        super().__init__()
+        self.codec_id = "jpeg"
+        self.single_channel = single_channel
+
+    def encode_single_image(self, image: np.ndarray) -> bytes:
+        """
+        Encode single image.
+
+        Example:
+            img = np.ones(100, 100, 3)
+            img_encoded = jpeg_codec.encode_single_image(img)
+
+        Args:
+            image (np.ndarray): Single image to be encoded
+
+        Returns:
+            Encoded data.
+        """
+        with BytesIO() as buffer:
+            Image.fromarray(image).save(
+                buffer, format=self.codec_id, subsampling=0, quality=90
+            )
+            return buffer.getvalue()
+
+    def decode_single_image(self, buf) -> np.ndarray:
+        """
+        Decode single image from buffer.
+
+        Example:
+            imgs_decoded = jpeg_codec.decode(imgs_encoded)
+
+        Args:
+            bytes_ (bytes): Encoded image
+
+        Returns:
+            Decoded data.
+        """
+        with BytesIO(buf) as buffer:
+            buffer.seek(0)
+            return np.array(Image.open(buffer, mode="r"))
+
+    @classmethod
+    def from_config(cls, config):
+        return JpegCodec(config["single_channel"])
+
+
+class PngCodec(BaseImgCodec, Codec):
+    def __init__(self, single_channel: bool = True):
+        """
+        Initialize PNG compressor
+
+        Args:
+            single_channel (bool): if True, encoder will remove the last dimension of input if it is 1.
+        """
+        super().__init__()
+        self.codec_id = "png"
+        self.single_channel = single_channel
+
+    def encode_single_image(self, image: np.ndarray) -> bytes:
+        """
+        Encode single image.
```

Comment:
if you are going to have the first line of every docstring repeat the name of the function but without the underscores, you may as well delete the line. although, it would be helpful if you elaborated on the function's utility/responsibility

Reply:
this is also confusing because you say "encode_single_image", but you aren't actually calling "self.encode", this is another reason we should rename "self.encode" -> "self.compress"